src/components/plots/plot.ts

```diff
@@ -224,18 +208,21 @@ module Plottable {
      *
      * Note that this will return all of the data attributes, which may not perfectly align to svg attributes
      *
-     * @param {datasetKey} the key of the dataset to generate the dictionary for
+     * @param {Dataset} dataset The dataset to generate the dictionary for
      * @returns {AttributeToAppliedProjector} A dictionary mapping attributes to functions
      */
-    public generateProjectors(datasetKey: string): AttributeToAppliedProjector {
-      var attrToProjector = this._generateAttrToProjector();
-      var plotDatasetKey = this._key2PlotDatasetKey.get(datasetKey);
-      var plotMetadata = plotDatasetKey.plotMetadata;
-      var userMetadata = plotDatasetKey.dataset.metadata();
+    public generateProjectors(dataset: Dataset): AttributeToAppliedProjector {
       var attrToAppliedProjector: AttributeToAppliedProjector = {};
-      d3.entries(attrToProjector).forEach((keyValue: any) => {
-        attrToAppliedProjector[keyValue.key] = (datum: any, index: number) => keyValue.value(datum, index, userMetadata, plotMetadata);
-      });
+      var datasetKey = this._keyForDataset(dataset);
+      if (datasetKey != null) {
+        var attrToProjector = this._generateAttrToProjector();
+        var plotDatasetKey = this._key2PlotDatasetKey.get(datasetKey);
```

Comment:
Doesn't really matter for the user / it's only an architectural complaint, but we really can just have a _dataset2PlotDatasetKey (or make that even cleaner)